gammapy/modeling/models/spatial.py

```diff
@@ -775,6 +775,49 @@ def to_region(self, **kwargs):
             **kwargs,
         )
 
+    @classmethod
+    def from_region(cls, region, **kwargs):
+        """Create a `DiskSpatialModel from a ~regions.EllipseSkyRegion`
+
+        Parameters
+        ----------
+        region : `~regions.EllipseSkyRegion` or ~regions.CircleSkyRegion`
+            region to create model from
+
+        Returns
+        -------
+        spatial_model : `~gammapy.modeling.models.DiskSpatialModel`
+        """
+        if isinstance(region, CircleSkyRegion):
+            region = EllipseSkyRegion(
+                center=region.center, width=region.radius, height=region.radius
+            )
+        if not isinstance(region, EllipseSkyRegion):
+            raise ValueError(
+                f"Please provide a `CircleSkyRegion` "
+                f"or `EllipseSkyRegion`, got {type(region)} instead."
+            )
+        if "frame" in kwargs:
+            frame = kwargs.pop("frame")
+            width = np.max(
+                [region.height.to("deg").value, region.width.to("deg").value]
+            )
+            wcs = WcsGeom.create(
+                skydir=region.center, binsz=0.02, frame=frame, width=width * u.deg
+            ).wcs
+            region = region.to_pixel(wcs).to_sky(wcs)
+        model = cls.from_position(region.center)
+        if region.height > region.width:
+            major_axis, minor_axis = region.height, region.width
+            phi = region.angle
+        else:
+            minor_axis, major_axis = region.height, region.width
+            phi = 90 * u.deg + region.angle
+        model.r_0.quantity = major_axis / 2.0
+        model.e.value = np.sqrt(1.0 - np.power(minor_axis / major_axis, 2))
+        model.phi.quantity = phi
+        return model
```

Comment:
Here are some more advanced ideas: e.g. it is usually a good idea to work with a pattern like:

```python3
kwargs.setdefault("r_0", major_axis / 2.0)
kwargs.setdefault("phi", phi)

return cls.from_position(region.position, **kwargs)
```
This way you could even overwrite quantities of the disk model.

For the frame handling you could maybe introduce a general utility method `region_to_frame()` in `gammapy.utils.regions` (and maybe make a feature request in `regions`). For an arbitrary region you can actually skip the computation of the width. The code here would simplify to:

```python3
frame = kwargs.pop("frame", region.center.frame)
region = region_to_frame(region, frame=frame)
```